deepchecks/analytics/anonymous_telemetry.py

```diff
@@ -14,29 +14,117 @@
 No credentials, data, personal information or anything private is collected (and will never be).
 """
 import http.client
+import json
 import os
 import pathlib
-import uuid
+import platform
+
 import deepchecks
+from deepchecks import is_notebook
+from deepchecks.analytics.utils import get_telemetry_config
 
 MODULE_DIR = pathlib.Path(__file__).absolute().parent.parent
-ANALYTICS_DISABLED = os.environ.get('DISABLE_DEEPCHECKS_ANONYMOUS_TELEMETRY', False)
+ANALYTICS_ENABLED = get_telemetry_config()['telemetry_enabled']
+
+
+def _identify_runtime():
+    """Identify the runtime."""
+    # If in docker
+    try:
+        with open('/proc/1/cgroup', 'rt', encoding='utf8') as f:
+            info = f.read()
+        if 'docker' in info or 'kubepods' in info:
+            return 'docker'
+    except (FileNotFoundError, Exception):  # pylint: disable=broad-except
+        pass
+
+    # If in colab
+    if 'COLAB_GPU' in os.environ:
+        return 'colab'
+
+    # If in notebook
+    if is_notebook():
+        return 'notebook'
+
+    # If in paperspace
+    if 'PAPERSPACE_NOTEBOOK_REPO_ID' in os.environ:
+        return 'paperspace'
+
+    return 'native'
+
+
+try:
+    RUNTIME = _identify_runtime()
+except Exception:  # pylint: disable=broad-except
+    RUNTIME = 'unknown'
+
 
+def get_environment_details():
+    """Get environment details."""
+    return {
+        'python_version': platform.python_version(),
+        'os': platform.system(),
+        'deepchecks_version': deepchecks.__version__,
+        'runtime': RUNTIME,
+    }
 
-def send_anonymous_import_event():
-    """Send an anonymous import event to PostHog."""
-    if not ANALYTICS_DISABLED:
+
+def send_anonymous_event(event_name, event_data=None):
+    """Send an anonymous event ."""
+    if ANALYTICS_ENABLED:
         try:
-            if os.path.exists(os.path.join(MODULE_DIR, '.user_id')):
-                with open(os.path.join(MODULE_DIR, '.user_id'), 'r', encoding='utf8') as f:
-                    user_id = f.read()
-            else:
-                user_id = str(uuid.uuid4())
-                with open(os.path.join(MODULE_DIR, '.user_id'), 'w', encoding='utf8') as f:
-                    f.write(user_id)
-
-            conn = http.client.HTTPSConnection('api.deepchecks.com', timeout=3)
-            conn.request('GET', f'/metrics?version={deepchecks.__version__}&uuid={user_id}')
+            machine_id = get_telemetry_config()['machine_id']
+            params = {
+                'event_name': event_name,
+                'event_data': event_data,
+                'machine_id': machine_id,
+            }
+            params.update(get_environment_details())
```

Comment:
why not put those properties under "env" key?

Reply:
I though it might be easier not to have a nested json (for queries, and BI tools)